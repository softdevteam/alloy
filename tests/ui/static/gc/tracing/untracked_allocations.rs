#![deny(untracked_heap_allocation)]
#![feature(allocator_api)]
#![feature(gc)]

use std::alloc::System;
use std::rc::Rc;

static A: System = System;

fn main() {
    let rc = Rc::new_in(123, A); //~ ERROR: uses a non-Alloy allocator
}
