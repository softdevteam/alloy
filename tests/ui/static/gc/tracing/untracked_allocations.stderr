error: uses a non-Alloy allocator
  --> $DIR/untracked_allocations.rs:11:14
   |
LL |     let rc = Rc::new_in(123, A);
   |              ^^^^^^^^^^^^^^^^^^
   |
   = help: Disable this lint only if you are certain this not contain a pointer to a GC object
note: the lint level is defined here
  --> $DIR/untracked_allocations.rs:1:9
   |
LL | #![deny(untracked_heap_allocation)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

